app/utils/ViewUtils.scala

```diff
@@ -24,4 +25,11 @@ object ViewUtils {
 
   def positiveOrZero(currentAmount: Amount): Amount =
     currentAmount.copy(amount = currentAmount.amount.max(BigDecimal(0)))
+
+  def positiveOrZero(currentRate: Rate):Rate={
+    if (currentRate.percent.head==('-'))
+      Rate.empty
+    else
+      currentRate
+  }
```

Comment:
I think instead of putting this logic in tax-summaries-frontend we can change the Rate we are getting from backend i.e. tax-summaries
https://github.com/hmrc/tax-summaries/blob/master/app/transformers/ATSCalculations.scala#L253
I think this is where from we are getting final percentage. If we can do something like 
val nicsRate = liabilityAsPercentage(nicsAndTaxPerCurrency)
    if (nicsRate.percent < 0) Rate(Rate.empty) else nicsRate 
then i think we would not need to do any change on the frontend. Make sure that we are not breaking any existing functionality.

Reply:
Not sure, I see what you are saying though. However the fix is following a pattern established when dealing with negative figures we get back for the Amount which you see in the view (nics.scala.html line 68) and in the ViewUtils Line 26 which zeros out negative numbers. This code is consistent with established display logic. The second aspect is that by putting the fix in the backend we are in effect changing the calculation to fit around the display logic, I don't know if we should do that, the backend is just responsible for getting the data, and performing the calculations and serving it to the frontend. It is up to the frontend to apply display logic with them? There could be a situation in the future where we call the backend and actually want the negative figures? I did think about if it should be in the backend and I certainly think there is a case for it, but i don't know if is actually the right place for it. So i followed precedent.

Reply:
I understand your point of view and I agree that this code is consistent with established display logic but if there is a way to improve it then we can have a look.
For second aspect my understanding is that we are already doing it for almost every field in PAYE by changing the calculation to return Amount as zero to fit around the display logic
https://github.com/hmrc/tax-summaries/blob/master/app/models/paye/PayeAtsData.scala#L63
I think backend is returning incorrect data by returning negative percentage of total income as income tax to the frontend and i think this should be the part of calculation logic on backend. 
We can discuss it if you want :)

Reply:
As discussed with Ste this is fine . In future we might need the negative figures from backend.
Happy to approve :)